types/underscore/index.d.ts

```diff
@@ -5674,16 +5708,24 @@ declare module _ {
         values(): _Chain<any>;
 
         /**
-        * Wrapped type `object`.
-        * @see _.mapObject
-        **/
-        mapObject(fn: _.ListIterator<T, any>): _Chain<T>;
+         * Like map, but for objects. Transform the value of each property in
+         * turn.
+         * @param iteratee The iteratee to use to transform property values.
+         * @param context `this` object in `iteratee`, optional.
+         * @returns A chain wrapper around a new object with all of the wrapped
+         * object's property values transformed through `iteratee`.
+         */
+        mapObject<I extends ObjectIteratee<V, any>>(
+            iteratee: I,
+            context?: any
+        ): _Chain<ObjectIterateeResult<I, TypeOfDictionary<V>, never>, { [K in keyof V]: ObjectIterateeResult<I, V[K]> }>;
```

Comment:
It might be nice as part of the TS3.0 changes to update the collection iterator type for non-dictionary-objects from `never` to `unknown` and loosen restrictions a bit on what constitutes a Dictionary since it's totally possible to use Collection functions on any object whether or not all of its properties have the same value type, it's just not as straightforward to type and `any` as a fallback seems a bit permissive.

I should note that passing a non-dictionary object to chain and calling Collection functions on it was already not great before I started in on this; previously it would have been picked up by the `chain<T extends {}>(obj: T): _Chain<T>;` overload, which would take the object type and make it the element type.

Reply:
I agree. One of the future items I'm planning is to swap `T` and `V` in `Underscore` and `_Chain`, which you [expressed no strong opinion around](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45201#issuecomment-660636917), but that was probably because I didn't do a good job of explaining what that would get us. That would allow us to provide a default value for `T` of `TypeOfCollection<V>`, which would mean that one would only have to specify a single type (the wrapped value type) when using `Underscore` and `_Chain`. I tried this out [over here](https://github.com/reubenrybnik/DefinitelyTyped/compare/chain-fixes-and-tests...reubenrybnik:underscore-value-centric-interfaces) way back at the beginning of this project just to see what would be involved and if there might be a way to maintain some sort of backwards compatibility with the original versions of `Underscore` and `_Chain`.

I might actually even get rid of `T` altogether and just use `TypeOfCollection<V>` and `TypeOfList<V>` directly in the collection and list functions since it would be nice for list-only functions to wind up with `never` as the iterable type when the wrapped object is a dictionary.